docs/package-deletion-policies.md

```diff
@@ -0,0 +1,210 @@
+
+When a package is deleted from a package manager, it frees the namespace that the package was occupying.
+Unless specifically prohibited, another package could be published with the name of the previously deleted package.
+Within a fairly conservative package deletion policy, this is unlikely to be a problem as the deleted package did not have widespread adoption.
+However, if a package manager allows for the deletion of widely used, established packages, the open namespace can become a security issue if a new (potentially malicious) package is published with the same namespace.
+Even if the new package isn't malicious, it has the potential to be confusing to users.
+Therefore, when considering a package deletion policy, it is important to consider what to do with the vacated namespaces.
+Some package managers never reuse a namespace, which is a reasonable security measure.
+Alternatively namespaces could be reused, with additional protections in place (for example, file hashing to notify users of changes).
+
+## Package Manager Structure and Package Deletion Policies
+
+Additionally, a package deletion policy could be impacted by architectural choices within the given package manager.
+
+### Decentralized Publishing
+
+For example, Go uses [decentralized publishing](https://go.dev/doc/modules/developing#decentralized) to make their modules available. 
+A Go module is published by tagging the code in its repository.
+When a module is published it is also uploaded to Go's [module proxy server](https://proxy.golang.org/).
+The module maintainer never directly adds their module to the proxy server, and there is not a way for the maintainer to delete the module from the proxy server.
+They can delete the original repository, but a copy of the project will remain on the proxy server.
+Instead, Go uses a [retract directive](https://go.dev/ref/mod#go-mod-file-retract), which is an example of a [package deletion alternative](#deletion-alternatives).
+
+### Test Instances and Staging Environments
+
+Another architecture decision related to a package deletion policy is the use of a separate test instance or a staging environment.
+
+For example PyPI maintains [TestPyPI](https://test.pypi.org/) to allow users to try distribution tools and processes without affecting the real PyPI.
+A test instance does not necessarily eliminate the need for a package deletion policy because mistakes can still be made, but it does allow for users to experiment without leaving junk packages on the main instance.
+
+## Deletion Alternatives
+
+There are a number of reasons that a person would want to delete a package.
+Package deletion is a good solution for some of those reasons (publishing the wrong package, making a typo, experimenting with the package manager), but not necessarily for others.
+
+What if a package (or version series) is no longer in active development?
+What if a specific version has a security issue?
```

Comment:
```suggestion
What if a specific version has a security issue or a severe bug?
```

We have received many deletion requests for a bugged release. We generally recommend unlisting (seems akin to yanking, as you have described below).